File: debugger/src/main/java/dev/appkr/debuggerex/HelloController.java
package dev.appkr.debuggerex;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.ResponseBody;

@Controller
public class HelloController {

    @GetMapping("canIDrive/{name}/{age}")
    @ResponseBody
    public String hello(@PathVariable String name, @PathVariable Integer age) {
        return notifyUserOfDriverStatus(name, age);
    }

    private String notifyUserOfDriverStatus(String name, Integer age) {
        String message = isOfDrivingAge(age) ? "You may drive" : "You man not drive";
        return name + ": " + message;
    }

    private Boolean isOfDrivingAge(Integer age) {
        return age >= 16;
    }

}

File: jib-docker/src/main/kotlin/kr/appkr/dockerdemo/HelloController.kt
package kr.appkr.dockerdemo

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.ResponseBody

@Controller
class HelloController {

    @GetMapping("/hello")
    @ResponseBody
    fun helloWorld(): String {
        return "Hello World"
    }

    @GetMapping("/hello/{name}")
    fun helloName(@PathVariable name: String, model: Model): String {
        model.addAttribute("name", name);
        return "hello_name" // Thymeleaf 템플릿 이름
    }

}
